File: src/questions/parkinglot/py/strategy/fee/HourlyFeeStrategy.py
from questions.py.entity import ParkingTicket
from questions.py.strategy.fee.FeeStrategy import FeeStrategy


class HourlyFeeStrategy(FeeStrategy):
    def __init__(self, hourly_rate: float):
        self._hourly_rate = hourly_rate

    def calculate_fee(self, ticket: ParkingTicket) -> float:
        return ticket.get_duration_in_hours() * self._hourly_rate

File: src/questions/parkinglot/java/strategy/fee/HourlyFeeStrategy.java
package questions.parkinglot.java.strategy.fee;

import questions.parkinglot.java.entity.ParkingTicket;

public class HourlyFeeStrategy implements FeeStrategy {

    private final float hourlyFee;

    public HourlyFeeStrategy(float hourlyFee) {
        this.hourlyFee = hourlyFee;
    }

    @Override
    public double calculateFee(ParkingTicket ticket) {
        return ticket.getTotalParkingDuration() *  hourlyFee;
    }
}
